db/main.go

```diff
@@ -9,15 +10,36 @@ import (
 
 var (
 	log = logging.Logger("router")
+	DB  *gorm.DB
 )
 
+type SpdTenantSvc struct {
+	DB         *gorm.DB
+	DryRunMode bool
+	Config     config.TenantServiceConfig
+}
+
+func NewSpdTenantSvc() *SpdTenantSvc {
```

Comment:
It's strange to me that this function is inside the `db` subpackage, IMO it should live elsewhere, as it contains more than just the DB struct but also the config (and may have more things added to it in the future). It's more like a global app context object. 

In the past I've put that in a subpackage called `core`, i.e, at the top level directory `/core/tenant-svc.go`

Reply:
In this `db` package, you should just have the stuff that opens the database and nothing to do with InitConfig parsing etc